src/org/opendatakit/briefcase/model/FormStatus.java

```diff
@@ -54,6 +55,13 @@ public void clearStatusHistory() {
 
   public void setStatusString(String statusString, boolean isSuccessful) {
     this.statusString = statusString;
+    if (statusHistory.length() > historyMaxSize) {
+      statusHistory.delete(0, statusString.length() + 1);
```

Comment:
It took me a bit to understand the delete bounds so just to have a record -- the first `+ 1` is needed to take into account the newline that will go with the new status. The second `+ 1` is because `lineEnd` is an index, not a length.